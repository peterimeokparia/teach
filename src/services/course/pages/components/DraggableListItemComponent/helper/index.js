import {
getSortedRecordsByPosition } from 'services/course/selectors';

export const getSortedRecordsBasedOnPosition = ( itemCollection ) => {
    return getSortedRecordsByPosition( itemCollection );
};

export const getCummulativePointsForCorrectlySortedItems = ( studentsAnswerCollection, answerKeyyCollection ) => {
    let cummulativePoints = 0;
  
    studentsAnswerCollection?.map( (item, index) => {

        if ( item.position === answerKeyyCollection[ index ]?.position ) {
            cummulativePoints += item.points;
        }
    });

    return cummulativePoints;
};