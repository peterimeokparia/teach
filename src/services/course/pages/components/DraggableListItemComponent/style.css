/* .item {
    width: 500px;
    margin-left: 600px;
   background-color: white;
} */

li:nth-child(odd) {
    /* width: 500px; */
    /* margin-left: 600px; */
    /* background-color: lightblue; */
}



/* li:nth-child(odd) {
    width: 500px;
    margin-left: 600px;
    background-color: lightblue;
} */