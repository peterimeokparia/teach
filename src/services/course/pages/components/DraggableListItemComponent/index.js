import { useState, useEffect } from 'react';
import { DragDropContext, Droppable, Draggable } from 'react-beautiful-dnd';
import { getSortedRecordsByPosition } from 'services/course/selectors';
import './style.css';

const DraggableListItemComponent = ({ 
  draggableListItemProps,
  children }) => {
  let {
    itemCollection,
    handleDraggableOnElementMove,
  } = draggableListItemProps;

    const [ updateContent, setUpdateContent ] = useState( false );

    useEffect(() => {
      if ( updateContent ) {
        setUpdateContent( false );
      }
    }, [ updateContent ] );

    let temp = [];

  function handleOnDragEnd(result) {
    if ( itemCollection?.length === 0 || undefined ) return;
    if (!result.destination) return;
    const items = Array.from(itemCollection);
    const [ reorderedItem ] = items.splice(result.source.index, 1);

    items.splice(result.destination.index, 0, reorderedItem);
      items.forEach( ( element, index ) => { 
        let repositionedItem = { ...element, position: ( index + 1 ) };

        handleDraggableOnElementMove( repositionedItem );
        temp.push( repositionedItem );
        setUpdateContent( true );
      }); 
  }

  return( 
    <div>
    <DragDropContext onDragEnd={handleOnDragEnd}>
      <Droppable droppableId="updatedOrderedQuestions">
        {(provided) => (
          <div className="listItem">
            <ul className="lessons" { ...provided.droppableProps } ref={ provided.innerRef }>
            {  getSortedRecordsByPosition( itemCollection )?.map((element, index) => { return( <Draggable key={element?._id} draggableId={element?._id} index={index}>
              {(provided) => (
                <li className={'lesson-item2'} id={element?._id} ref={provided.innerRef} {...provided.draggableProps} {...provided.dragHandleProps}>
                  {
                    children( element )            
                  }  
                  </li>
                )}
                </Draggable> 
              );
             } 
            )} 
            {provided.placeholder}
        </ul>
        </div>
        )}
      </Droppable>
    </DragDropContext>
    </div>
  );
};

export default DraggableListItemComponent;
  
      