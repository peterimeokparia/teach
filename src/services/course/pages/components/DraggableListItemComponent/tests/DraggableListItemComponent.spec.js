import React from 'react';
import DraggableListItemComponent from 'services/course/pages/components/DraggableListItemComponent';
import renderer from 'react-test-renderer';

describe('DraggableListItemComponent', () =>  {  

  let draggableListItemProps = {
    itemCollection: [ {_id: '001'}, {_id: '002'}, {_id: '003'} ],
    saveAction:()=>{}
  };

  it('renders as expected', async () => {
    const tree = renderer.create(
        <DraggableListItemComponent 
          draggableListItemProps={draggableListItemProps}
        >
          { ( element ) => {
                return <div>{ element?._id }</div>
              }
          }
        </DraggableListItemComponent > 
    );
      expect(tree.toJSON()).toMatchSnapshot();
  })
});




