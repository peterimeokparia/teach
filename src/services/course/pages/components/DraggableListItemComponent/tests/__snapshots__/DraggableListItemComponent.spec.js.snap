// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DraggableListItemComponent renders as expected 1`] = `
<div>
  <div
    className="listItem"
  >
    <ol
      className="lessons"
      data-rbd-droppable-context-id="0"
      data-rbd-droppable-id="updatedOrderedQuestions"
    >
      <li
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        className="lesson-item2"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="001"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="001"
        draggable={false}
        id="001"
        onDragStart={[Function]}
        onTransitionEnd={null}
        role="button"
        style={
          Object {
            "transform": null,
            "transition": null,
          }
        }
        tabIndex={0}
      >
        <div>
          001
        </div>
      </li>
      <li
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        className="lesson-item2"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="002"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="002"
        draggable={false}
        id="002"
        onDragStart={[Function]}
        onTransitionEnd={null}
        role="button"
        style={
          Object {
            "transform": null,
            "transition": null,
          }
        }
        tabIndex={0}
      >
        <div>
          002
        </div>
      </li>
      <li
        aria-describedby="rbd-hidden-text-0-hidden-text-0"
        className="lesson-item2"
        data-rbd-drag-handle-context-id="0"
        data-rbd-drag-handle-draggable-id="003"
        data-rbd-draggable-context-id="0"
        data-rbd-draggable-id="003"
        draggable={false}
        id="003"
        onDragStart={[Function]}
        onTransitionEnd={null}
        role="button"
        style={
          Object {
            "transform": null,
            "transition": null,
          }
        }
        tabIndex={0}
      >
        <div>
          003
        </div>
      </li>
    </ol>
  </div>
</div>
`;
